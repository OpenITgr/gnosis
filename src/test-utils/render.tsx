/* eslint-disable import/export */
import React, { FC, ReactElement } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options?: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
