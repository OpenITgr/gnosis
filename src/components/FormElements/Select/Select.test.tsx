import React from "react";
import userEvent from "@testing-library/user-event";
import { Select } from "./Select.stories";
import { screen, render } from "@test-utils/render";

describe("<Select />", () => {
  it("renders correctly", () => {
    render(<Select {...Select.args} />);

    const select = screen.getByLabelText("Choose a programming language");
    const options = screen.getAllByRole("option");

    expect(select).toBeInTheDocument();
    expect(options).toHaveLength(6);
    expect(options[0]).toHaveTextContent("Rust");
    expect(options[1]).toHaveTextContent("JavaScript");
    expect(options[2]).toHaveTextContent("TypeScript");
    expect(options[3]).toHaveTextContent("GoLang");
    expect(options[4]).toHaveTextContent("Python");
    expect(options[5]).toHaveTextContent("PHP");
  });

  it("selects a value", () => {
    render(<Select {...Select.args} />);

    const select = screen.getByLabelText("Choose a programming language");
    const options = screen.getAllByRole("option");

    expect(select).toHaveValue("rs");

    userEvent.selectOptions(select, options[1]);

    expect(select).toHaveValue("js");
  });

  it("matches snapshot", () => {
    const { container } = render(<Select {...Select.args} />);

    expect(container).toMatchSnapshot();
  });
});
