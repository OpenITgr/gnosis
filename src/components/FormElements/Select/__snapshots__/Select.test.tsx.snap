// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> matches snapshot 1`] = `
<div>
  <div
    class=" css-mqijmr-inputContainerBaseStyles-inputContainerBaseStyles-inputBaseStyles-inputBaseStyles-selectContainer-selectContainer"
  >
    <label
      class="css-xs6o3g-label-label"
      for="programming-language"
    >
      Choose a programming language
    </label>
    <div
      class="select-input-wrapper"
    >
      <select
        id="programming-language"
      >
        <option
          value="rs"
        >
          Rust
        </option>
        <option
          value="js"
        >
          JavaScript
        </option>
        <option
          value="ts"
        >
          TypeScript
        </option>
        <option
          value="go"
        >
          GoLang
        </option>
        <option
          value="python"
        >
          Python
        </option>
        <option
          disabled=""
          value="php"
        >
          PHP
        </option>
      </select>
      <span
        class="carret-wrapper"
      >
        <svg-file-stub
          height="24"
        />
      </span>
    </div>
  </div>
</div>
`;
